pkg/config/cluster/cluster.go

```diff
@@ -11,18 +11,33 @@ import (
 	"github.com/talos-systems/talos/pkg/crypto/x509"
 )
 
+// Name defines subset of Cluster to provide cluster name.
+type Name interface {
```

Comment:
I did this split-the-big-interface trick here to provide a way to have smaller mock in the test for Cluster config (in fact part of Cluster config required for cert generation)